File: KT10/src/main/java/net/println/kt10/LazyThreadSafeSynchronized.java
package net.println.kt10;

/**
 * Created by luliju on 2017/7/8.
 */
public class LazyThreadSafeSynchronized {
    private static LazyThreadSafeSynchronized INSTANCE;

    private LazyThreadSafeSynchronized(){}

    public static synchronized LazyThreadSafeSynchronized getInstance(){
        if(INSTANCE == null){
            INSTANCE = new LazyThreadSafeSynchronized();
        }
        return INSTANCE;
    }
}

File: KT10/src/main/kotlin/net/println/kt10/LazyThreadSafeSynchronized.kt
package net.println.kt10.kotlin

/**
 * Created by luliju on 2017/7/8.
 */
class LazyThreadSafeSynchronized private constructor() {
    companion object {
       private var instance: LazyThreadSafeSynchronized? = null

        @Synchronized
        fun get(): LazyThreadSafeSynchronized{
            if(instance == null) instance = LazyThreadSafeSynchronized()
            return instance!!
        }
    }

}